### ISSUE TITLE
[Bug] XF.UWP CollectionView memoryleak 

### ISSUE THREAD

**Issue by sjorsmiltenburg** [xamarin/Xamarin.Forms#15717]
### Description

Hi, is [this fix](https://github.com/dotnet/maui/pull/14075) for MAUI also going to be added to Xamarin.Forms?

I recently [reported a bug](https://developercommunity.visualstudio.com/t/XamarinForms-UWP-CollectionView-Memory/10319497) that is so similar that it is probably the same cause / fix.

### Steps to Reproduce

1. See [Bugreport](https://developercommunity.visualstudio.com/t/XamarinForms-UWP-CollectionView-Memory/10319497) for reproducable sample project on XF.UWP

### Expected Behavior
no memoryleak
### Actual Behavior
memoryleak
### Basic Information

- Version with issue:
- Last known good version:
- Platform Target Frameworks: 
  - iOS:  
  - Android:  
  - UWP:   
- Android Support Library / AndroidX Version: 
- NuGet Packages:
- Affected Devices:

### Environment

<details><summary>Show/Hide Visual Studio info</summary>

```
```

</details>

**Comment by jfversluis** [comment]
Will see if I can do that. If it's important to you, happy to have your PR as well!

**Comment by MitchBomcanhao** [comment]
just a sidenote - In your repro project you're using a xamarin forms package (5.0.0.2196) from october 2021. is it still as bad on the current version?

**Comment by sjorsmiltenburg** [comment]
@MitchBomcanhao yes, with an update to the latest version 5.0.0.2578 the problem still persists.
With a few resizes the mem usage of the sample app grows from 100 mb to 300+ and beyond.
I have updated the repo.

**Comment by sjorsmiltenburg** [comment]
> Will see if I can do that. If it's important to you, happy to have your PR as well!

The problem is quite pressing for me. I would love to submit a PR, I tried to invest some time into it and getting the framework to build/run on my machine takes me too much time. (I'm soo busy with other stuff and i'm quite out of my comfort zone here). So for me it almost makes more sense to invest my time in migrating to Maui as to create a PR to fix this problem for myself.
